go-controller/pkg/ovndbmanager/ovndbmanager.go

```diff
@@ -185,21 +191,14 @@ func ensureClusterRaftMembership(db string, kclient kube.Interface) {
 		}
 		knownServers = append(knownServers, server[1])
 	}
-	out, stderr, err := appCtl("cluster/status", dbName)
+	out, stderr, err := appCtl(5, "cluster/status", dbName)
 	if err != nil {
 		klog.Warningf("Unable to get cluster status for: %s, stderr: %v, err: %v", db, stderr, err)
-		if atomic.LoadInt32(clusterStatusRetryCnt) > maxClusterStatusRetry {
-			//delete the db file and start master
-			resetRaftDB(db)
-			atomic.StoreInt32(clusterStatusRetryCnt, 0)
-		} else {
-			atomic.AddInt32(clusterStatusRetryCnt, 1)
-			klog.Infof("Failed to get cluster status for: %s, number of retries: %d", db, *clusterStatusRetryCnt)
-		}
+		updateDBRetryCounter(dbRetryCnt, db)
 		return
 	}
 	// on retrieving cluster/status successfully reset the retry counter.
-	atomic.StoreInt32(clusterStatusRetryCnt, 0)
+	atomic.StoreInt32(dbRetryCnt, 0)
```

Comment:
If one of the checks in `ensureLocalRaftServerID()` failed, we basically just ignore the failure by resetting the retry count here if `cluster/status` succeeded.   While testing this PR I had replaced `cluster/sid` with `cluster/foo` expecting `resetRaftDB()` to be called eventually.  That didn't happen.  I guess I don't really understand the point of the `dbRetryCnt`.

Reply:
actually I take back what I said. It looks like how the code was written we are specifically only caring about if there was a failure to access the db here across all checks and then resetting the db/counter based on that. The code is totally not obvious about this. I'm going to add a commit that hopefully makes that more clear.